### ISSUE TITLE
Create pipeline that lints code for consistent practices

### ISSUE THREAD

**Issue by victorsandru** [Entropy675/CChess#59]
this pipeline should check for:

- no object files (if they are should they be deleted?)
- syntax errors
- naming conventions (UpperCamelCase for classes, lower_snake_case for functions, etc.)
- unit test output

**Comment by Entropy675** [comment]
yes, you should delete object files if they are there (make clean command does that)
and for the naming conventions its camelCase for functions and UpperCamelCase for classes, I'm not the most reasonable person so some names might not make much sense, if they are egregious enough you can change them when you find them.
